### PR TITLE
docs: add initial README for the abortable broadcast

### PR DESCRIPTION
**Pull request by rumenov**

### CODE REVIEW

**Review by yapignolet — APPROVED** [review]

File: rs/p2p/consensus_manager/README.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+:toc:  
+
+== Overview  
+
+This library implements the https://arxiv.org/abs/2410.22080[_Abortable Broadcast_] primitive.  
+
+Currently, the implementation has some limitations that restrict its generalization:  
+
+- **ArtifactAssembler coupling**: The implementation is tightly coupled with the `ArtifactAssembler`, reducing modularity.  
+- **Slot table deletions**: Deletions in the slot table are not propagated, which could lead to inconsistencies.
```
Comment (DSharifi) [inline rs/p2p/consensus_manager/README.adoc:11]: Why the "could"?

```suggestion
- **Slot table deletions**: Deletions in the slot table are not propagated, which lead to inconsistencies.  
```

File: rs/p2p/consensus_manager/README.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+[cols="3,3,3", options="header"]  
+|===  
+| **Capability** | **Purpose** | **Implementation Requirements**  
+
+| **PubSub-like library**  
+| Simplifies application development by abstracting one-to-many communication within a set of nodes.  
+| The library must implement https://en.wikipedia.org/wiki/Eventual_consistency[_eventual consistency_].  
+
+| **Multitenancy of applications**  
+| Allows multiple applications to safely use the library, ensuring isolation and uninterrupted operation for all.
```
Comment (DSharifi) [inline rs/p2p/consensus_manager/README.adoc:24]:
```suggestion
| Allows multiple applications to safely use the library, ensuring isolation and uninterrupted operation between all applications.  
```

File: rs/p2p/consensus_manager/README.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+| Allows multiple applications to safely use the library, ensuring isolation and uninterrupted operation for all.  
+| Must implement fair scheduling across peers to ensure resource allocation and isolation.  
+
+| **Operation in P2P networks with Byzantine peers**  
+| Ensures reliable message publishing and creation rates, even in networks with faulty or malicious nodes.  
+| The implementation must provide a non-IO-bound API and be memory-bounded.  
+
+| **Scalable and resource-efficient**  
+| Optimizes usage of shared resources (e.g., CPU, file descriptors) to maintain performance under high demand and prevent resource exhaustion.  
+| The library should expose a synchronous interface but implement asynchronous operations under the hood.
```
Comment (DSharifi) [inline rs/p2p/consensus_manager/README.adoc:33]: Not sure I follow. How do we provide a synchronous interface?

**Review by DSharifi — APPROVED** [review]

File: rs/p2p/consensus_manager/README.adoc (hunk 3 above)
Comment (rumenov) [reply rs/p2p/consensus_manager/README.adoc:33]: You can have a sync interface that doesn't do any IO but just mutates states and schedules async work. This is what we actually have.

**Pull request merged**